lib/lib.go

```diff
@@ -519,8 +519,13 @@ func (k *Khan) defaultPostRequest(
 }
 
 // SearchClans returns clan summaries for all clans that contain the string "clanName".
-func (k *Khan) SearchClans(ctx context.Context, clanName string) (*SearchClansResult, error) {
-	route := k.buildSearchClansURL(clanName)
+func (k *Khan) SearchClans(ctx context.Context, clanName string, useTextSearchOpt ... bool) (*SearchClansResult, error) {
+	useTextSearch := true
+	if len(useTextSearchOpt) > 0 {
+		useTextSearch = useTextSearchOpt[0]
```

Comment:
Why do you have an array of booleans if you only use the first one?

Reply:
The idea was to have an optional boolean argument, so as to be backwards compatible. Would it be better to disregard this?

Reply:
hmm, if you want backwards compatibility, maybe add a new method?

Reply:
Done.